src/lib.rs

```diff
@@ -213,6 +244,23 @@ pub enum Operation<K, V> {
     Empty(K),
     /// Remove all values in the value set for this key.
     Clear(K),
+    /// Retains all values matching the given predicate.
+    Retain(K, Predicate<V>),
+    /// Shrinks a value-set to it's minimum necessary size, freeing memory
+    /// and potentially improving cache locality if the `smallvec` feature is used.
+    ///
+    /// If no key is given, all value-sets will shrink to fit.
+    Fit(Option<K>),
+    /// Truncates the value-set for this key to the given length.
+    Truncate(K, usize),
+    /// Reverses the value-set for this key.
+    Reverse(K),
```

Comment:
`Reverse` and `Truncate` strike me as odd. I sort of see how they're analogous to the operations you can do on regular Rust collections, but for a value-set they seem somewhat strange. I wonder if we want to provide more "semantic" operations like `Pick(Front|Back, usize)`?

Reply:
After some thought I agree with you. We keep using the term "value-set", so perhaps we should restrict operations to those found in sets, like `HashSet`? More or less.

I mostly added `Reverse` and `Truncate` because they were easy and simple to expose, but they aren't necessary. It's probably best to not include those unless it becomes necessary. Reverse by itself especially isn't necessary, if the reader simply iterates in reverse.

Reply:
Okay, yeah, let's remove `Reverse` and `Truncate` for the time being then, and instead add them if there should be demand for them later?